### PR TITLE
Add sorting to validation error summary page

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

```
As a... an analyst
I need to... order validation errors
So that... I can see what the most common errors in past month/week are
```

## Changes proposed in this pull request

- Add a sorting dropdown to the page
- Update SQL query on back end to change sorting as required by the front end

## Guidance to review

- Pagination will be separate ticket
- The sorting is unit tested. Do we need to feature test this as well or is that overkill?
- Might need some design input on this - I [borrowed from Find](https://www.find-postgraduate-teacher-training.service.gov.uk/results?l=2&subjects%5B%5D=31&sortby=0) as inspiration.

sorting

## Link to Trello card

https://trello.com/c/0U8yaXyp/2725-add-ordering-pagination-to-validation-error-summary

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by davidgisbey — APPROVED** [review]
Tested it locally and it's working beautifully 👌

File: app/queries/validation_error_summary_query.rb [hunk 1]
```diff
@@ -43,13 +61,17 @@ class ValidationErrorSummaryQuery
       COUNT(DISTINCT user_id) AS unique_users_all_time
     FROM validation_error_counts
     GROUP BY form_object, attribute
-    ORDER BY incidents_all_time DESC".freeze
+    #{order}".freeze
```
Comment (stevehook) [inline app/queries/validation_error_summary_query.rb:64]: Does `.freeze` make sense as this is dynamic now? (I'm not sure it made sense before).

File: app/helpers/select_options_helper.rb [hunk 2]
```diff
@@ -23,6 +23,16 @@ def select_provider_options(providers)
     ] + providers.map { |provider| OpenStruct.new(id: provider.id, name: "#{provider.name} (#{provider.code})") }
   end
 
+  def select_sort_options
+    [
+      [0, 'All time'],
```
Comment (stevehook) [inline app/helpers/select_options_helper.rb:28]: Could the values here be strings, e.g. `all_time` to make the URL params more transparent?

File: app/helpers/select_options_helper.rb [hunk 3]
```diff
@@ -23,6 +23,16 @@ def select_provider_options(providers)
...
+      [1, 'Last week'],
+      [2, 'Last month'],
+    ].map do |sort_option|
```
Comment (stevehook) [inline app/helpers/select_options_helper.rb:31]: Make the (frozen) array a constant?

**Review by stevehook — APPROVED** [review]
Looks good and works well locally. Some minor queries inline.

File: app/queries/validation_error_summary_query.rb (hunk 1 above)
Comment (TobyRet) [reply app/queries/validation_error_summary_query.rb:64]: Yeah that can go - good spot

File: app/helpers/select_options_helper.rb (hunk 3 above)
Comment (TobyRet) [reply app/helpers/select_options_helper.rb:31]: I've extracted to a var to make it clearer. Not so sure about extracting to a constant and putting into an initialiser right now? Maybe if it's used in another place perhaps?

File: app/helpers/select_options_helper.rb (hunk 2 above)
Comment (TobyRet) [reply app/helpers/select_options_helper.rb:28]: Yep good idea

**Pull request merged**